File: Cynosura.Studio.Core/Requests/Solutions/Models/SolutionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using Cynosura.Studio.Generator;

namespace Cynosura.Studio.Core.Requests.Solutions.Models
{
    public class SolutionModel
    {
        public SolutionModel(string name, string path)
        {
            Name = name;
            Path = path;
        }

        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel? CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel? ModificationUser { get; set; }

        [DisplayName("Name")]
        public string Name { get; set; }

        [DisplayName("Path")]
        public string Path { get; set; }

        public string TemplateName { get; set; }

        public string TemplateVersion { get; set; }

        public void LoadMetadata()
        {
            var accessor = new SolutionAccessor(Path);
            TemplateName = accessor.Metadata.TemplateName;
            TemplateVersion = accessor.Metadata.TemplateVersion;
        }
    }
}

File: Cynosura.Studio.Web.Ng/src/app/solution-core/solution.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { SolutionFilter } from './solution-filter.model';

export class Solution {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  name: string;
  path: string;
  templateName: string;
  templateVersion: string;
}

export class SolutionListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new SolutionFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}
